odtp-app.sh

```diff
@@ -1,72 +1,141 @@
+        python3 /odtp/odtp-component-client/logger.py >> /odtp/odtp-logs/odtpLoggerDebugging.txt 2>&1 &
+    else
+        echo "'ODTP_MONGO_SERVER' does not exist or 'ODTP_LOGS_IN_DB' not set to 'TRUE'"
+    fi
+    return 0
+}
+
+# transfer input to output to s3
+function transfer_input_to_output() {
+    if [ "${TRANSFER_INPUT_TO_OUTPUT:-}" == "TRUE" ]; then
+        odtp::print_info \
+            "Copying input files '/odpt/odtp-input' to '/odtp/odpt-output'."
+        cp -r /odtp/odtp-input/* /odtp/odtp-output
+        if [ "$exit_code" -ne 0 ]; then
+            odtp::print_error \
+                "Coping input to output failed with exit code '$exit_code'."
+        fi        
+    fi
+    return 0
+}
 
 #  Take output and export it
-cd /odtp/odtp-output
-zip -rq ../odtp-output.zip ./*
-mv ../odtp-output.zip odtp-output.zip
-
-#########################################################
-# ODTP SNAPSHOT CREATION 
-#########################################################
-
-# Take snapshot of workdir
-if [[ "${ODTP_SAVE_SNAPSHOT:-}" = "TRUE" ]]; then
-    cd /odtp/odtp-workdir
-    zip -rq ../odtp-snapshot.zip ./*
-    mv ../odtp-snapshot.zip odtp-snapshot.zip
-fi
-
-## Saving Snapshot in s3
-
-if [[ -v ODTP_S3_SERVER && -v ODTP_MONGO_SERVER ]]; then
-    echo "Uploading to ODTP_S3_SERVER because 'ODTP_S3_SERVER' and 'ODTP_MONGO_SERVER' are both set"
-    python3 /odtp/odtp-component-client/s3uploader.py 2>&1 | tee /odtp/odtp-logs/odtpS3UploadedDebugging.txt  
-else
-    echo "Not uploading to S3 since either 'ODTP_S3_SERVER' or 'ODTP_MONGO_SERVER' is not specified"
-fi
-
-## Copying logs into output
-
-if [[ "${ODTP_LOGS_AS_FILE:-}" == "TRUE" ]]; then
-    echo "Writing log files to output since 'ODTP_LOGS_AS_FILE' is set to 'TRUE'"
-    cp /odtp/odtp-logs/log.txt /odtp/odtp-output/log.txt
-fi
+function compress_output () {
+    odtp::print_info "cd into '/odtp/odtp-output'"
+    cd /odtp/odtp-output
+    if [ "$exit_code" -ne 0 ]; then
+        odtp::print_error \
+            "'cd /odtp/odtp-output' failed with exit code '$exit_code'."
+    fi    
+    odtp::print_info "'zip -rq ../odtp-output.zip ./*'"
+    zip -rq ../odtp-output.zip ./*
+    if [ "$exit_code" -ne 0 ]; then
+        odtp::print_error \
+            "'zip -rq ../odtp-output.zip ./*' failed with exit code '$exit_code'."
+    fi      
+    odtp::print_info "cd into 'mv ../odtp-output.zip odtp-output.zip'"
+    mv ../odtp-output.zip odtp-output.zip
+    if [ "$exit_code" -ne 0 ]; then
+        odtp::print_error \
+            "'mv ../odtp-output.zip odtp-output.zip' failed with exit code '$exit_code'."
+    fi      
+    return 0
+}
+
+function compress_workdir () {
+    if [[ "${ODTP_SAVE_SNAPSHOT:-}" = "TRUE" ]]; then
+        odtp::print_info "'cd /odtp/odtp-workdir'"
+        cd /odtp/odtp-workdir
+        if [ "$exit_code" -ne 0 ]; then
+            odtp::print_error \
+                "'cd /odtp/odtp-workdir' failed with exit code '$exit_code'."
+        fi          
```

Comment:
suggestiion: Please do not glue together all lines. Thats kind of spagetti and hard to read.
The function is good however. Error reporting is much better, and its an improvement.

The error messages are however quite verbose, its ok for me, I would rather however keep it a little shorter.
Because it reads very hard, since  we have each command duplicated, once in the info, once the real call.
Also this gets outdated pretty soon, as you need to keep the real call and the message up to date. I would not want to maintain this code =).


Maybe better:

```shell
print_error "Failed to execute zip."
```

You any way need to look into the script if something fails. IMO